### PR TITLE
[feat]: [CDS-92269]: Added GitX Params Support To Harness-Go_Sdk For Infrastructures

### PR DESCRIPTION
**Pull request by senjucanon2**
## Describe your changes:
- Added Support for creating/updating/Importing remote Infrastructures via terraform.

## Testing:
- Validated creation of Infras in different scopes without GitX params.
- Created Remote Infrastructures in different Scopes.
- Imported Remote Infrastructures in different Scopes.
- Added Functional UTs.

## Creation of Remote Infrastructure:
Screenshot 2024-03-07 at 8 52 47 PM

## Creation of Inline Infrastructures:
Screenshot 2024-03-07 at 8 57 40 PM

## Comment Triggers

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- gitleaks: `trigger gitleaks`

### CODE REVIEW

File: harness/nextgen/api_infrastructures.go [hunk 1]
```diff
@@ -32,12 +32,29 @@ InfrastructuresApiService Create an Infrastructure in an Environment
  * @param ctx context.Context - for authentication, logging, cancellation, deadlines, tracing, etc. Passed from http.Request or context.Background().
  * @param accountIdentifier Account Identifier for the Entity.
  * @param optional nil or *InfrastructuresApiCreateInfrastructureOpts - Optional Parameters:
-     * @param "Body" (optional.Interface of InfrastructureRequest) -  Details of the Infrastructure to be created
+     * @param "Branch" (optional.String) -
```
Comment (sourabh-awashti) [inline harness/nextgen/api_infrastructures.go:35]: why did you remove body from comments??

File: harness/nextgen/api_infrastructures.go [hunk 2]
```diff
@@ -57,7 +74,33 @@ func (a *InfrastructuresApiService) CreateInfrastructure(ctx context.Context, ac
 	localVarFormParams := url.Values{}
 
 	localVarQueryParams.Add("accountIdentifier", parameterToString(accountIdentifier, ""))
-	// to determine the Content-Type header
+	if localVarOptionals != nil && localVarOptionals.Branch.IsSet() {
```
Comment (sourabh-awashti) [inline harness/nextgen/api_infrastructures.go:77]: don;t remove already added comments

File: harness/nextgen/api_infrastructures.go [hunk 3]
```diff
@@ -347,6 +406,15 @@ func (a *InfrastructuresApiService) GetInfrastructure(ctx context.Context, infra
...
 	}
+	if localVarOptionals != nil && localVarOptionals.Branch.IsSet() {
+		localVarQueryParams.Add("branch", parameterToString(localVarOptionals.Branch.Value(), ""))
+	}
+	if localVarOptionals != nil && localVarOptionals.RepoName.IsSet() {
+		localVarQueryParams.Add("repoName", parameterToString(localVarOptionals.RepoName.Value(), ""))
+	}
+	if localVarOptionals != nil && localVarOptionals.LoadFromFallbackBranch.IsSet() {
+		localVarQueryParams.Add("loadFromFallbackBranch", parameterToString(localVarOptionals.LoadFromFallbackBranch.Value(), ""))
+	}
```
Comment (sourabh-awashti) [inline harness/nextgen/api_infrastructures.go:417]: please add load from cache as well here

File: harness/nextgen/docs/InfrastructuresApi.md [hunk 4]
```diff
@@ -202,3 +203,44 @@ Name | Type | Description  | Notes
...
+ **optional** | ***InfrastructuresApiUpdateInfrastructureOpts** | optional parameters | nil if no parameters
+
+### Optional Parameters
+Optional parameters are passed through a pointer to a InfrastructuresApiUpdateInfrastructureOpts struct
+Name | Type | Description  | Notes
+------------- | ------------- | ------------- | -------------
+
+ **orgIdentifier** | **optional.String**|  | 
+ **projectIdentifier** | **optional.String**|  | 
+ **infraIdentifierIdentifier** | **optional.String**|  |
```
Comment (sourabh-awashti) [inline harness/nextgen/docs/InfrastructuresApi.md:225]: please correct the name here

File: harness/nextgen/api_infrastructures.go (hunk 2 above)
Comment (senjucanon2) [reply harness/nextgen/api_infrastructures.go:77]: it is shifted to Line 105

File: harness/nextgen/docs/InfrastructuresApi.md (hunk 4 above)
Comment (senjucanon2) [reply harness/nextgen/docs/InfrastructuresApi.md:225]: Done

Comment (sourabh-awashti) [reply harness/nextgen/docs/InfrastructuresApi.md:225]: infraIdentifierIdentifie I still see the same name

Comment (senjucanon2) [reply harness/nextgen/docs/InfrastructuresApi.md]: Done

**Review by sourabh-awashti — APPROVED** [review]
